File: app/src/main/java/com/booboot/vndbandroid/bean/vndbandroid/Status.java
package com.booboot.vndbandroid.bean.vndbandroid;

/**
 * Created by od on 15/03/2016.
 */
public class Status {
    public final static int UNKNOWN = 0;
    public final static int PLAYING = 1;
    public final static int FINISHED = 2;
    public final static int STALLED = 3;
    public final static int DROPPED = 4;
    public final static String DEFAULT = "Add to my VN list";

    public static String toString(int status) {
        switch (status) {
            case 0:
                return "Unknown";
            case 1:
                return "Playing";
            case 2:
                return "Finished";
            case 3:
                return "Stalled";
            case 4:
                return "Dropped";
            default:
                return DEFAULT;
        }
    }
}

File: app/src/main/java/com/booboot/vndbandroid/model/vndbandroid/Status.kt
package com.booboot.vndbandroid.model.vndbandroid

import com.booboot.vndbandroid.App
import com.booboot.vndbandroid.R

object Status {
    const val UNKNOWN = 0
    const val PLAYING = 1
    const val FINISHED = 2
    const val STALLED = 3
    const val DROPPED = 4
    const val DEFAULT = "Add to my VN list"
    val ALL = listOf(PLAYING, FINISHED, STALLED, DROPPED, UNKNOWN)

    fun toString(status: Int?) = when (status) {
        0 -> "Unknown"
        1 -> "Playing"
        2 -> "Finished"
        3 -> "Stalled"
        4 -> "Dropped"
        else -> null
    }

    fun toShortString(status: Int?): String = when (status) {
        0 -> "?"
        1 -> "P"
        2 -> "F"
        3 -> "S"
        4 -> "D"
        else -> App.context.getString(R.string.dash)
    }
}
